unittests/linuxliketests.py

```diff
@@ -317,6 +317,19 @@ def test_generate_gir_with_address_sanitizer(self):
         self.init(testdir, extra_args=['-Db_sanitize=address', '-Db_lundef=false'])
         self.build()
 
+    def test_qt5dependency_no_lrelease(self):
+        '''
+        Test that qt5 detection with qmake works. This can't be an ordinary
+        test case because it involves setting the environment.
+        '''
+        testdir = os.path.join(self.framework_test_dir, '4 qt')
+        def _no_lrelease(self, prog, *args, **kwargs):
+            if 'lrelease' in prog:
+                prog = 'lrelease-not-found'
+            return self._interpreter.find_program_impl(prog, *args, **kwargs)
```

Comment:
This does work, but you could also just return a `programs.NonExistingExternalProgram(prog)`, which would do the same thing, but without the potential of someone actually having a program called `lrelease-not-found` in their path.

Reply:
it seems better, I changed it